sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/TextSocketSourceProvider.scala

```diff
@@ -58,22 +57,16 @@ class TextSocketSourceProvider extends DataSourceV2
     new TextSocketTable(
       options.get("host").get,
       options.getInt("port", -1),
+      options.getInt("numPartitions", SparkSession.active.sparkContext.defaultParallelism),
```

Comment:
this is a small fix. The [test](https://github.com/apache/spark/pull/23619/files#diff-fa0715a30e2a82ca2c4297024e9441d8L298) needs to specify the `numPartitions`, but the socket source always use the spark default parallelism. Here I make `numPartitions` configurable.